rs/consensus/src/dkg/payload_validator.rs

```diff
@@ -111,141 +119,159 @@ pub(crate) fn validate_payload(
     validation_context: &ValidationContext,
     metrics: &IntCounterVec,
 ) -> ValidationResult<PayloadValidationError> {
+    let current_height = parent.height.increment();
+    let registry_version = pool_reader
+        .registry_version(current_height)
+        .expect("Couldn't get the registry version.");
```

Comment:
I guess it implicitly relies on the [signature of the proposal being checked first](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/consensus/src/consensus/validator.rs?L187), which would return an error if the registry version can't be determined. Maybe it would be cleaner to also return an error here.

Reply:
Thanks for the comment. I copied this line without thinking about it too much. 
As Leo said we are already checking the registry version before we enter this function, so we shouldn't panic here. Nevertheless, as Leo suggests, we should return an error here instead of relying on some assumptions

Reply:
Could there be a situation where we might be better of if we crash? For example, say, we fail to obtain the registry version for some reason. If we return an error we conclude that an otherwise valid payload is invalid (which might lead to getting stuck?). If we crash we have a chance that we can obtain the registry version after the restart and validate the payload.

Reply:
> If we return an error we conclude that an otherwise valid payload is invalid (which might lead to getting stuck?).

Returning an error here doesn't necessarily imply that the payload is invalid.
We are returning [ValidationError::ValidationFailed](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/interfaces/src/validation.rs?L11-12) here which means that we don't know if the artifact is valid or invalid. In practice it means we will keep retrying to validate the payload

Reply:
>If the expectation is that an error can either be an invalid payload or a "I don't know" an error would be an option, otherwise not.

We do have both error types:
- `InvalidDkgPayloadReason` for definitely invalid payloads, which means the artifact will be removed from the unvalidated section
- `DkgPayloadValidationFailure` for "I don't know", which means the artifact will remain in the unvalidated section, and we will retry the validation later.

This function looks at the latest summary block payload to extract the agreed registry version for the DKG interval. So receiving `None` means that `current_height` falls outside of the interval for that summary block. So intuitively I would say retrying makes sense, in the hope that the next time we do have the correct summary.

That being said we should actually never be in this position since we shouldn't be validating blocks that are below the summary or too far ahead. But I'm not sure if it makes sense to encode that assumption here

Reply:
Thanks a lot both. This is what I was missing.